core/arch/arm/plat-stm32mp1/main.c

```diff
@@ -281,18 +276,57 @@ early_init_late(set_all_gpios_non_secure);
 #endif /* CFG_STM32_GPIO */
 #endif /* CFG_STM32MP13 */
 
+#if defined(CFG_STM32MP15) && defined(CFG_TZSRAM_START)
+/*
+ * When SYSRAM is used as TZSRAM, TZSRAM shall fit either in SYSRAM secure
+ * address range or in the full SYSRAM and inside SRAMs address range.
+ * Configuration allows to locate TZSRAM in DDR for debug and test purpose.
+ */
+#define SYSRAM_END		(SYSRAM_BASE + SYSRAM_SIZE)
+#define TZSRAM_IS_IN_SYSRAM	((CFG_TZSRAM_START >= SYSRAM_BASE) && \
+				 (CFG_TZSRAM_START < SYSRAM_END))
+
+#define TZSRAM_ENDS_IN_SRAMS	((CFG_TZSRAM_START + CFG_TZSRAM_SIZE <= \
+				  SRAM4_BASE + SRAM4_SIZE) && \
+				 (SYSRAM_SEC_SIZE == SYSRAM_SIZE))
+
+#define TZSRAM_ENDS_IN_SYSRAM_SEC	(CFG_TZSRAM_START + CFG_TZSRAM_SIZE <= \
+					 SYSRAM_BASE + SYSRAM_SEC_SIZE)
+
+static_assert(!TZSRAM_IS_IN_SYSRAM || TZSRAM_ENDS_IN_SYSRAM_SEC ||
+	      TZSRAM_ENDS_IN_SRAMS);
+
+#if TZSRAM_IS_IN_SYSRAM
```

Comment:
What happens if TZSRAM covers SRAMs and not SYSRAM then?

Reply:
nice catch. i'll consider the case. thanks.